### PR TITLE
WIP: tool migration to Chibi-scheme

### PR DESCRIPTION
**Pull request by ekaitz-zarraga**
Migration of all the tools to chibi scheme.
Pull request for discussion and development

### CODE REVIEW

File: src_chibi/ksyscalls.scm [hunk 1]
```diff
@@ -75,88 +117,135 @@
...
+            ,(hash-table-ref wrapper 'vars)
+            ,(hash-table-ref wrapper 'commoncode) ,nl
+            ,(hash-table-ref wrapper 'execcode) ,nl
+            "}"
 
+            )))
 
+(define (default_syscall_ret-wrapper wrapper)
+  (joined displayed
+    `( "\tcase" ,(hash-table-ref wrapper 'id) ":" ,nl
```
Comment (ekaitz-zarraga) [inline src_chibi/ksyscalls.scm]: Use `(indent N)` function

**Review by ekaitz-zarraga — COMMENTED** [review]
- Added some support for elegant indentation not based on tabs and fixed to 4 spaces.
    - Have to make it coherent in the whole file
- Still not really sure about the use of `joined displayed` pair all the time. There should be a simpler way to solve it.
- Thinking about making a macro/function that creates simple codeblocks line by line from a list of lines to avoid having to use `indent` and `nl` so much. Just send lines in a list and let the func join everything together.

**Comment by ekaitz-zarraga** [comment]
First working version of `ksyscalls` in chibi-scheme, tested against an output c file generated by the go tools and has a 100% match in the output.

Please @Shamar take a look to this.

**Comment by Shamar** [comment]
Looks fine to me.

As for indentation, C code in Jehanne is indented with tabs (as in Plan 9). Feel free to indent scheme as you like, but I'd prefer to have all C code we are responsible for that follows the same conventions.

Also, please wait before attacking `usyscalls`: I've modified it in the process of porting GCC (because of... reasons...) and if you port the current version in master it would requires some adjustments later. I hope to complete the port in few weeks, so I hope it won't be a huge issue.

**Comment by ekaitz-zarraga** [comment]
>  As for indentation, C code in Jehanne is indented with tabs (as in Plan 9). Feel free to indent scheme as you like, but I'd prefer to have all C code we are responsible for that follows the same conventions.

UGH TABS ;)

Anyway I prepared the code to be able to work on that easily.
Just added a commit that solves that.

**Comment by Shamar** [comment]
Hi @ekaitz-zarraga [I finally moved Jehanne out of GitHub](http://jehanne.h--k.it/2022/01/06/jehanne_is_moving.html).

I close this issue, as I've also replaced Go with rc scripts.

Thanks for your help and support!

**Pull request closed without merging**